packages/sales-channel/src/services/sales-channel-module.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  Context,
+  DAL,
+  FilterableSalesChannelProps,
+  FindConfig,
+  InternalModuleDeclaration,
+  ISalesChannelModuleService,
+  ModuleJoinerConfig,
+  SalesChannelDTO,
+} from "@medusajs/types"
+import { InjectManager, MedusaContext } from "@medusajs/utils"
+import { CreateSalesChannelDTO, UpdateSalesChannelDTO } from "@medusajs/types"
+
+import { SalesChannel } from "@models"
+
+import { joinerConfig } from "../joiner-config"
+import SalesChannelService from "./sales-channel"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  salesChannelService: SalesChannelService<any>
+}
+
+export default class SalesChannelModuleService<
+  TEntity extends SalesChannel = SalesChannel
+> implements ISalesChannelModuleService
+{
+  protected baseRepository_: DAL.RepositoryService
+  protected readonly salesChannelService_: SalesChannelService<TEntity>
+
+  constructor(
+    { baseRepository, salesChannelService }: InjectedDependencies,
+    protected readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    this.baseRepository_ = baseRepository
+    this.salesChannelService_ = salesChannelService
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  @InjectManager("baseRepository_")
+  async create(
+    data: CreateSalesChannelDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<SalesChannelDTO[]> {
+    const salesChannel = await this.salesChannelService_.create(
+      data,
+      sharedContext
+    )
+
+    return this.baseRepository_.serialize<SalesChannelDTO[]>(salesChannel, {
+      populate: true,
+    })
+  }
+
+  @InjectManager("baseRepository_")
+  async delete(
+    ids: string[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    await this.salesChannelService_.delete(ids, sharedContext)
+  }
+
+  @InjectManager("baseRepository_")
```

Comment:
question: Should this be `@InjectTransactionManager("baseRepository_")`? cc @adrien2p

Reply:
yes, should be tx manager...fixed

Reply:
Yes it should be tx manager